### PR TITLE
Get repositories instead of templates from extensions

### PR DESCRIPTION
**Pull request by makandre**
Signed-off-by: Andrew Mak <makandre@ca.ibm.com>

Based on recent discussion with @tobespc, extensions will now return additional template repositories to Codewind (and let Codewind read the templates from it), rather than extensions doing the parsing and returning template objects to Codewind

This PR is to handle that scenario and is required by kabanero-io/appsodyExtension#18

### CODE REVIEW

**Comment by makandre** [comment]
Hi @rwalle61, sorry I somehow messed up my old branch & PR (https://github.com/eclipse/codewind/pull/97) so I'm resubmitting my changes under this PR.

This PR contains the unit tests that you requested.

File: test/src/unit/template.test.js [hunk 1]
```diff
@@ -10,6 +10,7 @@
 *******************************************************************************/
 const chai = require('chai');
 
+global.codewind = { RUNNING_IN_K8S: false };
```
Comment (rwalle61) [inline test/src/unit/template.test.js]: why's this needed? and shouldn't we unset it after the test (or don't use a global) to decouple tests?

File: src/pfe/portal/modules/Templates.js [hunk 2]
```diff
@@ -54,7 +54,32 @@ module.exports = class Templates {
     }
 
     let newProjectTemplates = [];
-    await Promise.all(this.repositoryList.map(async function getTemplates(repository) {
+
```
Comment (rwalle61) [inline src/pfe/portal/modules/Templates.js]: nit: I'd prefer this code in a subfunction, as this function is getting large

File: test/src/unit/template.test.js (hunk 1 above)
Comment (makandre) [reply test/src/unit/template.test.js]: Without this, there's an undefined error when loading the `Templates` class, which is due to https://github.com/eclipse/codewind/blob/master/src/pfe/portal/modules/utils/sharedFunctions.js#L19 trying the read the global

I'm not sure how the existing tests were able to run before, they too fail without this.

Setting `RUNNING_IN_K8S` to false explicitly is not needed tho, and I changed that.

File: src/pfe/portal/modules/Templates.js (hunk 2 above)
Comment (makandre) [reply src/pfe/portal/modules/Templates.js]: moved

**Comment by makandre** [comment]
@rwalle61 Ok, I removed the global.  The tests pass if they are all run during the build because of https://github.com/eclipse/codewind/blob/master/test/src/unit/projectInitializer.test.js#L20

(I was running `template.test.js` on its own hence I had to add that)

**Pull request merged**